services/graphService.js

```diff
@@ -58,28 +59,46 @@ GraphService.prototype.publicRankedList = function() {
 	return dfd.promise
 }
 
-GraphService.prototype.userGraphs = function(username) {
+GraphService.prototype.userGraphs = function(username, filter, offset, limit) {
 	var dfd = when.defer()
 
-	this._model
-		.find({ owner: username, deleted: false })
-		.select('_creator private owner name previewUrlSmall updatedAt stat')
+	filter = filter || {}
+	filter = _.extend({deleted:false}, filter)
+	filter.owner = username
+
+	var query = this._model
+		.find(filter)
 		.sort('-updatedAt')
-		.exec(function(err, list)
-	{
-		if (err)
-			return dfd.reject(err)
 
-		dfd.resolve(list)
-	})
+	if (offset)
+		query.skip(offset)
+
+	if (limit)
+	 	query.limit(limit)
+
+	query
+		.select('_creator private owner name previewUrlSmall previewUrlLarge updatedAt stat views')
+		.exec(function(err, list) {
+			if (err)
+				return dfd.reject(err)
+
+			dfd.resolve(list)
+		})
 
 	return dfd.promise
 }
 
-GraphService.prototype._save = function(data, user) {
+// e.g. when toggling public/private or renaming, but not touching anything else
+// opts {updatedAt, version}
+GraphService.prototype._save = function(data, user, opts) {
 	var that = this
 	var wasNew = false
 
+	opts = _.extend({
+		version : currentPlayerVersion,
```

Comment:
this should not be updated to the newest player in the case of public/private switching.